lib/Github/HttpClient/HttpClient.php

```diff
@@ -80,12 +99,25 @@ public function clearHeaders()
 
     public function addListener($eventName, $listener)
     {
-        $this->client->getEventDispatcher()->addListener($eventName, $listener);
+        // Guzzle 3 BC
+        if (method_exists($this->client, 'getEmitter')) {
+            $this->client->getEmitter()->on($eventName, $listener);
+        } else {
+            $this->client->getEventDispatcher()->addListener($eventName, $listener);
+        }
     }
 
-    public function addSubscriber(EventSubscriberInterface $subscriber)
+    /**
+     * @param EventSubscriberInterface|SubscriberInterface $subscriber
+     */
+    public function addSubscriber($subscriber)
     {
-        $this->client->addSubscriber($subscriber);
+        // Guzzle 3 BC
+        if (method_exists($this->client, 'getEmitter')) {
+            $this->client->getEmitter()->attach($subscriber);
```

Comment:
IMo, these methods should be deprecated in this class. Proxying methods of the interface implementation does not make sense IMO. The SDK should not leak the internal guzzle API